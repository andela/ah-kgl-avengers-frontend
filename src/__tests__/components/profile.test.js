import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Profile from '../../components/container/profile';
import ProfileEditor from '../../components/container/profileEditor';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  user: {
    user: {
      username: 'user',
    },
    profile: { username: '' },
    userArticles: [],
    isRequestOn: false,
  },
});

const props = {
  onGetUser: jest.fn(),
  onUpdateUser: jest.fn(),
  match: { params: { username: 'user' } },
  history: {
    push: jest.fn(),
  },
};

describe('<Profile/>', () => {
  beforeEach(() => {});

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('match the snapshot of the Profile component', () => {
    const component = shallow(
      <Provider store={store}>
        <Profile {...props} />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
  it('match the snapshot of the ProfileEditor component', () => {
    const component = shallow(
      <Provider store={store}>
        <ProfileEditor {...props} />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });

  it('should mount the Profile page', () => {
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <Profile {...props} />
        </Provider>
      </BrowserRouter>,
    );
    expect(wrapper.find('Profile')).toBeDefined();
  });

  it('should mount the ProfileEditor page', () => {
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <ProfileEditor {...props} />
        </Provider>
      </BrowserRouter>,
    );
    expect(wrapper.find('ProfileEditor')).toBeDefined();
  });
});
