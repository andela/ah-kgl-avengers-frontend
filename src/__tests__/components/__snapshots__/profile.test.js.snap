// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> match the snapshot of the Profile component 1`] = `ShallowWrapper {}`;

exports[`<Profile/> match the snapshot of the ProfileEditor component 1`] = `ShallowWrapper {}`;
